packages/vsx-registry/src/browser/vsx-extensions-model.ts

```diff
@@ -59,20 +74,71 @@ export class VSXExtensionsModel {
     @inject(VSXExtensionsSearchModel)
     readonly search: VSXExtensionsSearchModel;
 
-    protected readonly initialized = new Deferred<void>();
+    @inject(RequestService)
+    protected request: RequestService;
+
+    @inject(OVSXApiFilter)
+    protected vsxApiFilter: OVSXApiFilter;
 
     @postConstruct()
     protected init(): void {
-        this.doInit();
-    }
-
-    protected async doInit(): Promise<void> {
-        await Promise.all([
+        this.initialized = Promise.all([
             this.initInstalled(),
             this.initSearchResult(),
             this.initRecommended(),
-        ]);
-        this.initialized.resolve();
+        ]).then(() => { }, error => {
+            console.error(error);
+        });
```

Comment:
Minor: Should `catch` be more straightforward here?
```suggestion
        ]).catch(error => {
            console.error(error);
        });
```